app/models/terminal_executor.rb

```diff
@@ -75,6 +75,20 @@ def quiet
 
   private
 
+  def cancel(timeout:)
+    kill "INT"
+    timeout.times do
+      return unless kill(0) # stop when not running
```

Comment:
so you signal 0 each second? that seems odd...

Reply:
huh. TIL: https://stackoverflow.com/questions/11012527/what-does-kill-0-pid-in-a-shell-script-do

Reply:
maybe make a `pgid_running?` command that internally does `kill(0)` so it's more obvious?